core-bundle/src/Routing/Content/ContentUrlResult.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Content;
+
+use Contao\CoreBundle\Exception\ForwardPageNotFoundException;
+use Contao\PageModel;
+use Nyholm\Psr7\Uri;
+
+final class ContentUrlResult
+{
+    private bool $redirect = false;
+
+    public function __construct(public readonly object|string|null $content)
```

Comment:
```suggestion
    private function __construct(public readonly object|string|null $content)
```

I think we should make sure the developers only work with our static factories so we can update and adjust this as we like in the future.

Reply:
That won't work, because the `StringResolver` needs this atm. We would need to add more API to explicitly allow final string URLs, but not sure that's a better API. We could mark the method as `@internal` though maybe?

Reply:
Why not a `ContentUrlResult::fromStringUrl()`? Nobody else is going to use it and if they do, the API is still neat 🤔

Reply:
why would nobody use that? How do I know whether to use `ContentUrlResult::url()` or `fromStringUrl()`? But yes, it can be more explicit.

Reply:
Can't the `StringResolver` call the respective static factories depending on the content itself?

Reply:
Improved in a941ccd5f25836ab4949c0205c3a470605af3179